File: Backend/scoring_service/core/domain/apparatus.go
package domain

import (
	"encoding/json"
	"fmt"
)

type Apparatus int8

const (
	Floor Apparatus = iota
	PommelHorse
	StillRings
	Vault
	ParallelBars
	HorizontalBar
	BalanceBeam
	UnevenBars
)

type Apparatuses []Apparatus

func (a *Apparatuses) scanApparatuses(src interface{}) error {
	switch src := src.(type) {
	case Apparatuses:
		jsonData, err := json.Marshal(src)
		if err != nil {
			return err
		}
		return json.Unmarshal(jsonData, a)
	default:
		return fmt.Errorf("unsupported type for Apparatuses: %T", src)
	}
}

File: Frontend/Public/gco-public/src/app/model/core/apparatus.ts
export enum Apparatus{
    Floor = 0,
    PommelHorse = 1,
    StillRings = 2,
    Vault = 3,
    ParallelBars = 4,
    HorizontalBar = 5,
    BalanceBeam = 6,
    UnevenBars = 7,
}

export function getApparatusName(value: Apparatus): string {
    return Apparatus[value] || Apparatus[0];
}

